src/core/Akka.Persistence/Persistence.cs

```diff
@@ -81,15 +79,13 @@ public string PersistenceId(IActorRef actor)
         public IActorRef SnapshotStoreFor(string snapshotPluginId)
         {
             var configPath = string.IsNullOrEmpty(snapshotPluginId) ? _defaultSnapshotPluginId.Value : snapshotPluginId;
-            Lazy<PluginHolder> pluginContainer;
-            var extensionIdMap = _snapshotPluginExtensionIds.Value;
-            if (!extensionIdMap.TryGet(configPath, out pluginContainer))
-            {
-                pluginContainer = new Lazy<PluginHolder>(() => CreatePlugin(configPath, _ => DefaultPluginDispatcherId), LazyThreadSafetyMode.ExecutionAndPublication);
-                _snapshotPluginExtensionIds.CompareAndSet(extensionIdMap, extensionIdMap.AddOrUpdate(configPath, pluginContainer));
-                return SnapshotStoreFor(snapshotPluginId);
-            }
```

Comment:
It should be thread safe. CompareAndSet Sets only the new value if extensionIdMap is still the same. SnapshotStoreFor should return allways the valid value